test/e2e_node/hugepages_test.go

```diff
@@ -117,43 +118,89 @@ func makePodToVerifyHugePages(baseName string, hugePagesLimit resource.Quantity,
 	return pod
 }
 
-// configureHugePages attempts to allocate hugepages of the specified size
-func configureHugePages(hugepagesSize int, hugepagesCount int) error {
-	// Compact memory to make bigger contiguous blocks of memory available
-	// before allocating huge pages.
-	// https://www.kernel.org/doc/Documentation/sysctl/vm.txt
-	if _, err := os.Stat("/proc/sys/vm/compact_memory"); err == nil {
-		if err := exec.Command("/bin/sh", "-c", "echo 1 > /proc/sys/vm/compact_memory").Run(); err != nil {
-			return err
-		}
-	}
-
+// reserveHugePages just writes appropriate value into sysfs
+func reserveHugePages(hugepagesSize int, hugepagesCount int) error {
 	// Reserve number of hugepages
 	// e.g. /bin/sh -c "echo 5 > /sys/kernel/mm/hugepages/hugepages-2048kB/vm.nr_hugepages"
 	command := fmt.Sprintf("echo %d > %s-%dkB/%s", hugepagesCount, hugepagesDirPrefix, hugepagesSize, hugepagesCapacityFile)
 	if err := exec.Command("/bin/sh", "-c", command).Run(); err != nil {
 		return err
 	}
 
-	// verify that the number of hugepages was updated
-	// e.g. /bin/sh -c "cat /sys/kernel/mm/hugepages/hugepages-2048kB/vm.nr_hugepages"
-	command = fmt.Sprintf("cat %s-%dkB/%s", hugepagesDirPrefix, hugepagesSize, hugepagesCapacityFile)
-	outData, err := exec.Command("/bin/sh", "-c", command).Output()
-	if err != nil {
-		return err
+	if hugepagesCount > 0 {
+		// verify that the number of hugepages was updated
+		// e.g. /bin/sh -c "cat /sys/kernel/mm/hugepages/hugepages-2048kB/vm.nr_hugepages"
+		command = fmt.Sprintf("cat %s-%dkB/%s", hugepagesDirPrefix, hugepagesSize, hugepagesCapacityFile)
+		outData, err := exec.Command("/bin/sh", "-c", command).Output()
+		if err != nil {
+			return err
+		}
+
+		numHugePages, err := strconv.Atoi(strings.TrimSpace(string(outData)))
+		if err != nil {
+			return err
+		}
+
+		framework.Logf("Hugepages total is set to %v", numHugePages)
+		if numHugePages == hugepagesCount {
+			return nil
+		}
+
+		return fmt.Errorf("expected hugepages %v, but found %v", hugepagesCount, numHugePages)
 	}
+	return nil
+}
 
-	numHugePages, err := strconv.Atoi(strings.TrimSpace(string(outData)))
+func traceMemInfo() {
```

Comment:
IMHO we should add it under global AfterEach, something similar to https://github.com/kubernetes/kubernetes/blob/49f9e227e985bcbf6fd4781db1788b78a8e140af/test/e2e/framework/framework.go#L448
So it can be great to get:
1. Memory information
2. Top 10 memory consumers
3. Probably all running containers